File: 0027-remove-element/0027-remove-element.py
class Solution(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        indxs = []
        size = len(nums)
        for i in range(size):
            if nums[i] != val:
                indxs.append(i)
        k = len(indxs)
        for i in range(k):
            nums[i] = nums[indxs[i]]
        return k
        

File: 0027-remove-element/0027-remove-element.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int k = 0; // Pointer for the next position of a non-val element
        
        for (int i = 0; i < nums.length; i++) {
            // If the current element is not the value to remove
            if (nums[i] != val) {
                nums[k] = nums[i];
                k++;
            }
        }
        
        return k; // k is the count of elements not equal to val
    }
}